src/main.ts

```diff
@@ -4,24 +4,45 @@ import * as dotenvExpand from "dotenv-expand";
 import inputs from "./inputs";
 import * as core from "@actions/core";
 
+const DEFAULT_SEPARATOR = '|'
+
 function processValue(name: string, value: string) {
     if (inputs.mask) {
         core.setSecret(value)
     }
-
     if (inputs.export) {
         core.exportVariable(name, value)
     } else {
         core.setOutput(name, value)
     }
 }
 
+function readFile(name: string): dotenv.DotenvParseOutput {
+    return dotenv.parse(fs.readFileSync(name))
+}
+
+function getVars(): dotenv.DotenvParseOutput {
+    const separator = inputs.separator || DEFAULT_SEPARATOR
+    const files = inputs.envFile.split(separator)
+    return files.reduce((accum, file) => ({
+        ...accum,
+        ...readFile(file)
+    }), {})
+}
+
 export function runImpl() {
-    let vars = dotenv.parse(fs.readFileSync(inputs.envFile))
+    let vars = getVars()
     if (inputs.expand || inputs.expandWithJobEnv) {
-        // @ts-ignore
-        vars = dotenvExpand.expand({parsed: vars, ignoreProcessEnv: !inputs.expandWithJobEnv}).parsed
+        vars = dotenvExpand.expand({parsed: vars, ignoreProcessEnv: !inputs.expandWithJobEnv}).parsed as dotenv.DotenvParseOutput
+    }
+    
+    if (inputs.variables && inputs.variables.toLocaleLowerCase() !== 'all') {
```

Comment:
well...but every name is valid for a variable...you can also have a variable called `true="myValue"`
The intention here is...if variable param is not defined, or you use 'all', you retrieve all the variables and you will have access to them in the output. Usually you have both options to specify if you want to include all of them, but if you prefer to remove 'all', I can remove it, and the only option to generate an output with all the variables is to do not use variables definition.
This is up to you...I can remove it if you want

Reply:
One of the option is to call this input `filter` and:
1. if it's empty, it will allow all variables to be exported
2. if it's not empty it will work as a regexp that will allow variable names matching the regexp to be exported

It will:
1. implement the similar to file separators (`|`) but more powerful semantics (`|`  means "OR" in regexps)
2. keep backwards compatibility if the input not set

Share your thoughts about that

Reply:
Updated to RegExp, and remove all as a value